test/unit/sync_mediator_test.rb

```diff
@@ -118,18 +121,21 @@ def test_calling_instance_sync_does_not_allow_instances_to_affect_the_global_cac
     bad_citizen = BadCitizen.new('foo')
     stub_kubectl_response('get', 'BadCitizen', *@params, resp: { "items" => [bad_citizen.kubectl_response] }, times: 1)
 
-    mediator.sync([bad_citizen])
+    # Cache is only warmed if we batch fetch
+    mediator.sync([bad_citizen] * (KubernetesDeploy::SyncMediator::LARGE_BATCH_THRESHOLD + 1))
     assert_equal bad_citizen.kubectl_response, mediator.get_instance('BadCitizen', bad_citizen.name) # still cached
   end
 
   def test_sync_calls_sync_on_each_instance
-    stub_kubectl_response('get', 'FakeDeployment', *@params,
-      resp: { "items" => [@fake_deployment.kubectl_response] }, times: 1)
-    stub_kubectl_response('get', 'FakePod', *@params, resp: { "items" => [@fake_pod.kubectl_response] }, times: 1)
-    all_cm = [@fake_cm.kubectl_response, @fake_cm2.kubectl_response]
-    stub_kubectl_response('get', 'FakeConfigMap', *@params, resp: { "items" => all_cm }, times: 1)
+    test_resources = [@fake_pod, @fake_cm, @fake_deployment]
+    test_resources.each { |r| r.expects(:sync).once }
+    mediator.sync(test_resources)
+  end
+
+  def test_sync_does_not_warm_cache_with_few_resources
+    KubernetesDeploy::Kubectl.any_instance.expects(:run).with('get', 'FakeConfigMap', @fake_cm.name, *@params).never
```

Comment:
I guess it doesn't really matter because an exception will be raised if either request is attempted, but technically the cache warming would be requesting the list, not the instance.